packages/ra-ui-materialui/src/detail/Create.tsx

```diff
@@ -58,11 +59,19 @@ export const Create = (
 ): ReactElement => {
     useCheckMinimumRequiredProps('Create', ['children'], props);
     const controllerProps = useCreateController(props);
-    return (
+    const body = (
         <CreateContextProvider value={controllerProps}>
             <CreateView {...props} {...controllerProps} />
         </CreateContextProvider>
     );
+    return props.resource ? (
```

Comment:
I'v considered xtracting this duplicated logic to a dedicated commponent, like so:

```jsx
const ConditionalWrapper = ({ condition, wrapper, children }) => 
  condition ? wrapper(children) : children;

...

<ConditionalWrapper
   condition={props.resource}
   wrapper={children => <ResourceContextProvider value={props.resource}>{children}</ResourceContextProvider>}
>
    <CreateContextProvider value={controllerProps}>
        <CreateView {...props} {...controllerProps} />
    </CreateContextProvider>
</ConditionalWrapper>
```

But I find it less legible. Besides, the duplication is only a few lines.